test/hotspot/jtreg/compiler/c2/irTests/scalarReplacement/AllocationMergesTests.java

```diff
@@ -0,0 +1,932 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package compiler.c2.irTests.scalarReplacement;
+
+import compiler.lib.ir_framework.*;
+
+/*
+ * @test
+ * @bug 8281429
+ * @summary Tests that C2 can correctly scalar replace some object allocation merges.
+ * @library /test/lib /
+ * @requires vm.compiler2.enabled
+ * @run driver compiler.c2.irTests.scalarReplacement.AllocationMergesTests
+ */
+public class AllocationMergesTests {
+    private static Point global_escape = new Point(2022, 2023);
+
+    public static void main(String[] args) {
+        TestFramework.runWithFlags("-XX:+ReduceAllocationMerges",
+                                   "-XX:CompileCommand=exclude,*::dummy*");
+    }
+
+    // ------------------ No Scalar Replacement Should Happen in The Tests Below ------------------- //
```

Comment:
I was expecting to see a similar comment indicating when the tests that *do* result in Scalar Replacement start to appear. Although you've commented on some tests individually, from this comment here it implies that all tests below do not result in SR. I don't think that's the case though. Can you please review and update to make sure all non-SR-resulting tests and SR-resulting tests are contiguous and separated by comments?

Reply:
We talked on Teams about this.